### ISSUE TITLE
PacerRssFeed() throws away times, wrongly

### ISSUE THREAD

**Issue by johnhawkinson** [freelawproject/juriscraper#223]
One of the most valuable features of the RSS feed that's not available to general PACER users (but is available to filing CMECF users) is the timestamps of events. Down to the second.

But we just throw them away instead of preserving them:
https://github.com/freelawproject/juriscraper/blob/97fc853db3b8971af3e24f3b0562ab3346faf7fa/juriscraper/pacer/rss_feeds.py#L192-L201

That seems wrong on multiple levels. Even if the CL database doesn't support times in this field, which it appears not to:

https://github.com/freelawproject/courtlistener/blob/9137b4607a920b779354e1238b1b975e67aaa64e/cl/search/models.py?utf8=%E2%9C%93#L186

It's still a mistake for the parser to throw away this information. It's a _parser_, it doesn't pick and choose what to parse, it _parses_ it. (Apparently Django doesn't let you assign a DateTimeField to a DateField without [thinking](https://stackoverflow.com/questions/10289941/django-set-datefield-from-a-datetimefield) about it? Laaaaame.)

But more significantly, this field needs to migrate to a datetime. And furthermore it should be very clear that it is storing the time in the local timezone of the court (that information needs to be in the `models.py` `help_text`), and also what that time zone is (probably means a new field -- although that can be derived from the `court` field, that's going to be very confusing to anyone trying to deal with the data and a time should be a time, not a nonstandard thing offset by an unknowable).

We went to all this trouble to figure out what the timezone of every court is in a177072ce9b2841def1de5c53a419273cf1777df, although the commit message does not reflect the nature of the work done (subsequently [separately broken out ](https://github.com/johnhawkinson/ecftimezone)). We should use that information.

**Comment by johnhawkinson** [comment]
> But more significantly, this field needs to migrate to a datetime. And furthermore it should be very clear that it is storing the time in the local timezone of the court (that information needs to be in the models.py help_text), and also what that time zone is (probably means a new field -- although that can be derived from the court field, that's going to be very confusing to anyone trying to deal with the data and a time should be a time, not a nonstandard thing offset by an unknowable).

Err, wat? No.
I mean, there are some choices:
1. `date_filed` is a DateTime in the local timezone of the court.
2. `date_filed` is a DateTime in a the standard timezone, Coordinated Universal Time (UTC)
3. `date_filed` is a DateTime in some other timezone, like `US/Pacific` where Mike lives

It seems clear to me that (1), which I suggested in the first comment, really doesn't make sense. It would mean, for instance, that if you did a group of queries across multiple cases in CL, you could not sensibly order the results based on the DateTime field.

It also seems to me that (3) is...just unwise. But I may be speaking with East Coast parochialism.

Of course, it seems like no matter which of these choices are chosen, the zone needs to be recorded. Does it need to be recorded once for every DateTime field? I fear that's the best practice.

Timezone experts will tell you that events should be stored in the zone they are referenced to, so that when the zone rules change it doesn't leave you with a mess. It's what lead to my suggesting (1). But that implies a database type that stores timezones and ~we don't have that~ -- but actually we do!

We set `USE_TZ = True` in Django settings, and Python's `datetime` tuples support timezones.
Except...the Python API for this is very confusing and I can't even figure out how to use it.

So...I don't know.
Other than I think (3) is right out, because [the Django docs](https://docs.djangoproject.com/en/2.0/topics/i18n/timezones/) say:

> When support for time zones is enabled, Django stores datetime information in UTC in the database, uses time-zone-aware datetime objects internally, and translates them to the end user’s time zone in templates and forms.

So basically (2), except we need an additional zone field for every datetime field? Maybe? Probably? I don't even.

**Comment by mlissner** [comment]
Reading this over, and forgetting everything I've read about TZs in the past, doesn't this kind of tell us the answer? Whatever we throw at Django, it:

> stores datetime information in UTC in the database

And when we pull things from Django, it:

> uses time-zone-aware datetime objects internally, and translates them to the end user’s time zone in templates and forms.

I don't know *how* it possibly knows the end user's time zone, BUT I think this tells me that the important thing to do here is to set the TZ on the date object we save to the DB, because if we don't, it'll just assume UTC?

**Comment by mlissner** [comment]
This all makes me suspect that the real trick, for RSS feeds is indeed to use the Court-specific TZ info to set the TZ on our objects, and from there...profit?

**Comment by johnhawkinson** [comment]
> BUT I think this tells me that the important thing to do here is to set the TZ on the date object we save to the DB, because if we don't, it'll just assume UTC?

No, it doesn't say that.
It says we must do **either** of:

1. Set the correct TZ (court local tz) in the `datetime` object we store in the database.
2. Convert the time to UTC before storing it in the database.
3. ~Set the TZ to `US/Pacific` in the `datetime` object (and convert to `US/Pacific`) before we store in the database.~

But when I said 
>  Except...the Python API for this is very confusing and I can't even figure out how to use it.

above, it meant that I do not know how to do (1).

**Comment by mlissner** [comment]
This should be fixed by https://github.com/freelawproject/courtlistener/pull/2477.
